ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -813,6 +813,128 @@ export default {
         },
       },
     },
+    {
+      id: 'E12S Basic Relativity Yellow Hourglass',
+      // Orient where "Yellow" Anger's Hourglass spawns
+      netRegex: NetRegexes.addedCombatantFull({ npcNameId: '9824' }),
+      durationSeconds: 15,
+      infoText: (data, matches, output) => {
+        const y = parseFloat(matches.y) + 75;
+        const x = parseFloat(matches.x);
+
+        // Positions are the 8 cardinals + numerical slop on a radius=20 circle.
+        // Positions are also moved downward 75
+        // N = (0, -95), E = (20, -75), S = (0, -55), W = (-20, -75)
+        // NE = (14, -89), SE = (14, -61), SW = (-14, -61), NW = (-14, -89)
+        // Map N = 0, NE = 1, ..., NW = 7
+        const dirs = {
+          0: output.north(),
+          1: output.northeast(),
+          2: output.east(),
+          3: output.southeast(),
+          4: output.south(),
+          5: output.southwest(),
+          6: output.west(),
+          7: output.northwest(),
+        };
+
+        const dir = Math.round(4 - 4 * Math.atan2(x, y) / Math.PI) % 8;
+
+        return output.hourglass({
+          dir: dirs[dir],
+        });
+      },
+      outputStrings: {
+        north: Outputs.north,
+        northeast: Outputs.northeast,
+        east: Outputs.east,
+        southeast: Outputs.southeast,
+        south: Outputs.south,
+        southwest: Outputs.southwest,
+        west: Outputs.west,
+        northwest: Outputs.northwest,
+        hourglass: {
+          en: 'Yellow: ${dir}',
+        },
+      },
+    },
+    {
+      id: 'E12S Adv Relativity Hourglass Collect',
+      // Collect Sorrow's Hourglass locations
+      netRegex: NetRegexes.addedCombatantFull({ npcNameId: '9823' }),
+      run: (data, matches) => {
+        const id = parseInt(matches.id, 16);
+        const y = parseFloat(matches.y) + 75;
+        const x = parseFloat(matches.x);
+
+        // Positions are NE, N, NW, SW, S, SE + numerical slop on a radius=20 circle.
+        // Positions are also moved downward 75
+        // N = (0, -86), S = (0, -64)
+        // NE = (10, -80), SE = (10, -69), SW = (-10, -69), NW = (-10, -80)
+        // Map NW = 0, N = 1, ..., W = 7
+        const dir = Math.round(5 - 4 * Math.atan2(x, y) / Math.PI) % 8;
+
+        data.sorrows = data.sorrows || {};
+        data.sorrows[id] = [dir, false];
+      },
+    },
+    {
+      id: 'E12S Adv Relativity Hourglass Collect Yellow Tethers',
+      // '0086' is the Yellow tether that buffs "Quicken"
+      // '0085' is the Red tether that buffs "Slow"
+      netRegex: NetRegexes.tether({ id: '0086' }),
+      run: (data, matches) => {
+        const id = parseInt(matches.sourceId, 16);
+        data.sorrows[id][1] = true;
+      },
+    },
+    {
+      id: 'E12S Adv Relativity Speed',
+      // Orient where Oracle Quickens Sorrow's Hourglass
+      netRegex: NetRegexes.startsUsing({ id: '58DD' }),
```

Comment:
Is there a reason to have a second trigger here? Could all this be done in the `E12S Adv Relativity Hourglass Collect Yellow Tethers` above? It seems like as soon as you have one yellow tether, you know everything.

I think this might also let you simply `data.sorrows` to just be `data.sorrows[id] = dir;` in the collection step.

Reply:
I seem to have issues getting suppressSeconds to behave, but I changed up the method to use this. FWIW, the previous version did work in my testing. I'll be able to test the new code tomorrow.